src/Rules/Arrays/NonexistentOffsetInArrayDimFetchRule.php

```diff
@@ -96,13 +97,15 @@ static function (Type $type) use ($dimType): bool {
 			return [];
 		}
 
-		$hasOffsetValueType = $type->hasOffsetValueType($dimType);
+		$keyDimType = ArrayType::castToArrayKeyType($dimType);
```

Comment:
This isn't right, this rule is used for checking other offsets, not only arrays. That's why I suggested that `castToArrayKeyType` should be called in `ConstantArrayType::hasOffsetValueType()` and that's how this would be fixed.